fixtures/prov_filter.py

```diff
@@ -54,6 +64,11 @@ def parse_filter(cmd_filter):
     filtered_providers = provider_keys()
     for provider in provider_keys():
         data = cfme_data['management_systems'][provider]
+        ver_raw = data.get('restricted_version', None)
+        if ver_raw:
+            op, ver = re.search('([><=!]{1,2})\s*([\d\.]+)', ver_raw).groups()
+            if not mapping[op](current_version(), ver):
+                filtered_providers.remove(provider)
```

Comment:
For the purposes of readability and flexibility, I think a regex here may be avoidable.

This is a little more verbose, and despite the nested loop might end up being faster than the regex (not that I think it really matters here):
```python
    for provider in provider_keys():
        data = cfme_data['management_systems'][provider]

        restricted_version = data.get('restricted_version', None)
        if restricted_version:
            # _version_operator_map = mapping
            for op, comparator in _version_operator_map.items():
                # split string by op; if the split works, version won't be empty
                head, op, version = restricted_version.partition(op)
                if not version:
                    continue
                # alternative way to get version, if for some reason you hate partition
                # try:
                #     __, version = restricted_version.split(op)
                # except ValueError:
                #     continue
                if not comparator(current_version(), version):
                    filtered_providers.remove(provider)
                    break
            else:
                raise NoOperatorFoundOrSomethingException(restricted_version)

        tags = data.get('tags', [])
        if provider not in cmd_filter and not set(tags) & set(cmd_filter):
            filtered_providers.remove(provider)
```

The problem with this method is that now the mapping must be an ordereddict, and the more specific matches must come first: `>=` must come before `>`, for example, since `>=` starts with `>`.  In return, though, you get to raise an error if restricted_version isn't defined correctly, which is pretty sweet, plus nobody needs to mentally parse a regex to debug it.